NEWS.md

```diff
@@ -1,5 +1,10 @@
 # teal.data 0.6.0.9012
 
+### Deprecation
```

Comment:
Are we using a section like that? How it is done (possibly in other packages) up until now? I was about to call it a breaking change but it's soft deprecate so it's not necessarily a breaking type of a change

Reply:
I think we put soft-deprecation in breaking changes as well https://github.com/insightsengineering/teal/blob/main/NEWS.md?plain=1#L80

Reply:
ok so let's follow the convention and call it a breaking change then

Reply:
d2cec0ef4d82b62b9a751c7cf93cb76ea0002f50
thanks Pawel for pointing this out